tests/InstrEngineTests/NaglerInstrumentationMethod/NaglerInstrumentationMethod.cpp

```diff
@@ -1187,6 +1202,133 @@ HRESULT CInstrumentationMethod::PerformSingleReturnInstrumentation(IMethodInfo*
     return S_OK;
 }
 
+HRESULT CInstrumentationMethod::AddExceptionHandler(IMethodInfo* pMethodInfo, IInstructionGraph* pInstructionGraph)
```

Comment:
>AddExceptionHandler [](start = 32, length = 19)

Add a comment somewhere in here that this function doesn't handle multiple returns.

Reply:
Also, consider adding some checks to ensure that there isn't more than one return instruction, and failing if there is. Otherwise, the execution of the test may end up throwing an exception during the jit of the method, and it would be difficult to track down why.

---
In reply to: [566343023](https://github.com/microsoft/CLRInstrumentationEngine/pull/353#discussion_r566343023) [](ancestors = 566343023)

Reply:
Your tests don't cover it, but tail calls can be a problem as well. You cannot have a tail call instruction inside try/catch/finally blocks.

---
In reply to: [566348196](https://github.com/microsoft/CLRInstrumentationEngine/pull/353#discussion_r566348196) [](ancestors = 566348196,566343023)

Reply:
Added checks for this here: https://github.com/robertpi/CLRInstrumentationEngine/commit/3b9aedb3812b7f8fc56fc937dd0a9b02ecab0105